pkg/resource/document/deploy.go

```diff
@@ -35,15 +35,23 @@ import (
 	deployErrors "github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/deploy/errors"
 )
 
-//go:generate mockgen -source=document.go -destination=document_mock.go -package=document documentClient
-type Client interface {
+//go:generate mockgen -source=deploy.go -destination=document_mock.go -package=document DeploySource
+type DeploySource interface {
 	Get(ctx context.Context, id string) (documents.Response, error)
 	List(ctx context.Context, filter string) (documents.ListResponse, error)
 	Create(ctx context.Context, name string, isPrivate bool, externalId string, data []byte, documentType documents.DocumentType) (api.Response, error)
 	Update(ctx context.Context, id string, name string, isPrivate bool, data []byte, documentType documents.DocumentType) (api.Response, error)
 }
 
-func Deploy(ctx context.Context, client Client, properties parameter.Properties, renderedConfig string, c *config.Config) (entities.ResolvedEntity, error) {
+type DeployAPI struct {
+	source DeploySource
+}
+
+func NewDeployAPI(source DeploySource) *DeployAPI {
+	return &DeployAPI{source}
+}
+
+func (d DeployAPI) Deploy(ctx context.Context, properties parameter.Properties, renderedConfig string, c *config.Config) (entities.ResolvedEntity, error) {
```

Comment:
Some error cases are not handled in the unit tests. I know, you just moved the code. Maybe we can add some test cases if you agree?

- missing name parameter
- error reading received data
- multiple documents found with externalId
- failed to unmarshal dashboard payload